test/jdk/java/awt/Robot/RobotModifierMaskTest.java

```diff
@@ -0,0 +1,275 @@
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Robot;
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.StringSelection;
+import java.awt.event.KeyEvent;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.WindowConstants;
+
+import jtreg.SkippedException;
+import sun.awt.OSInfo;
+
+import static javax.swing.SwingUtilities.invokeAndWait;
+
+/*
+ * @test
+ * @bug 8302618
+ * @key headful
+ * @modules java.desktop/sun.awt
+ * @library /test/lib
+ * @requires (os.family == "mac")
+ * @summary To test if modifier keys work properly, during Robot's Key Event
+ *          with and without manual mouse move.
+ *
+ * @run main RobotModifierMaskTest
+ * @run main/manual RobotModifierMaskTest manual
+ */
+
+public class RobotModifierMaskTest {
+
+    private static Robot robot;
+    private static JFrame jFrame;
+    private static JTextArea jTextArea;
+    private static boolean startTest = false;
+    private static boolean isManual = false;
+
+    private static StringBuffer errorLog = new StringBuffer();
+    private static final String EXPECTED_RESULT_SHIFT = "AAAAA";
+    private static final String EXPECTED_RESULT_CAPS = "AaAaAa";
+    private static final String EXPECTED_RESULT_META = "AAA";
+    private static final String EXPECTED_RESULT_ALT = "\u00e5\u00e5\u00e5";
+    private static final String EXPECTED_RESULT_CTRL = "0";
+
+    private static final String INSTRUCTIONS = """
+            This test is running in manual mode to check the effect of typing modifier keys
+            through Robot combined with concurrent external manual mouse movement.
+            It tests the following key modifiers - Shift, Caps, Control, Option and Command.
+
+            In this mode when the Robot starts to type, the user is required to concurrently
+            move the mouse (without clicking/dragging).
+            When ready click on the "Start" button to run the test.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        if (OSInfo.getOSType() != OSInfo.OSType.MACOSX) {
+            throw new SkippedException("macOS test only");
+        }
+
+        try {
+            isManual = args.length != 0;
+
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+            robot.setAutoDelay(100);
+
+            // create instruction frame when running in manual mode
+            if (isManual) {
+                try {
+                    invokeAndWait(RobotModifierMaskTest::createInstructionsUI);
+                    robot.waitForIdle();
+                    while (!startTest) {
+                        robot.delay(200);
+                        if (!jFrame.isVisible()) {
+                            throw new RuntimeException("Test instruction frame closed");
+                        }
+                    }
```

Comment:
Instead of busy-waiting, you can use `CountDownLatch` to start the test. You should also release the latch if the instruction frame is closed.